src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -446,16 +446,42 @@ auto_flash(std::shared_ptr<xrt_core::device> & workingDevice)
   // Collect all indexes of boards need updating
   std::vector<std::pair<unsigned int , DSAInfo>> boardsToUpdate;
 
-  static boost::property_tree::ptree ptEmpty;
-  auto available_shells = pt.get_child(std::to_string(workingDevice->get_device_id()) + ".platform.available_shells", ptEmpty);
+  std::string image_path;
+  if(!image.empty()) {
+  //iterate over installed shells
+  //check the vbnv against the vnbv passed in by the user
+  auto installedShells = firmwareImage::getIntalledDSAs();
+  int multiple_shells = 0;
+    
+  for(auto const& shell : installedShells) {
```

Comment:
I like the fact that you are looking for duplicates of the shells, unfortunately the check for multiple shells is done later in the code flow.

Being that the code needs to look through the installedShells and that if multiple matches are unexceptionable, it would be better to move the line 469-470 to 460 so that the next developer KNOWS that this is not acceptable.